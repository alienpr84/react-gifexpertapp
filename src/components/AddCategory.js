import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');

	const handleOnChangeInputValue = (e) => {
		setInputValue(e.target.value);
	}

	const handleOnSubmit = (e) => {
		e.preventDefault();
		
		if(inputValue.trim().length > 2) {
			setCategories( prev => [inputValue, ...prev]);
			setInputValue('');
		}
		
	}

	return (
		<div>
			<form onSubmit={handleOnSubmit}>
				<input 
					value= { inputValue }
					onChange={ handleOnChangeInputValue }
					placeholder="Enter a value"
				/>
			</form>
		</div>
	)
}

AddCategory.prototype = {
	setCategories: PropTypes.func.isRequired,
}