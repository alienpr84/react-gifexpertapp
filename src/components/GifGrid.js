import useFetchGifs from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs( category )

  return (
    <div>
      <h3>{category}</h3>

			{ loading && 'loading...' }

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem
						key={img.id}
						{...img}
					/>
        ))}
      </div>
    </div>
  );
};

export default GifGrid;
