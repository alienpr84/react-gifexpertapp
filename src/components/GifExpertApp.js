import { useState } from 'react';
import { AddCategory } from './AddCategory';
import GifGrid from './GifGrid';

const GifExpertApp = () => {
	const [categories, setCategories] = useState(['One punch']);

  return (
    <>
      <h2>GifExpertApp</h2>
			<AddCategory setCategories={ setCategories } />
      <hr />

			<ul>
			{
				categories.map( category => <GifGrid key={ category } category={ category }/>)
			}
			</ul>
    </>
  );
};

export default GifExpertApp;
