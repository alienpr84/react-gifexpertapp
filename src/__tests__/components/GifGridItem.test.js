import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Testing in <GifGridItem />', () => {
	const props = { 
		title: 'One Punch',
		url: 'https://www.google.com'
	};
	const wrapper = shallow(<GifGridItem {...props} />);

  test('Must show <GifGridItem /> component correctly', () => {
		expect(wrapper).toMatchSnapshot();
  });

	test('Must have a url equal to: "https://www.google.com"', () => {
		const url = wrapper.find('img').prop('src');
		expect(url).toBe(props.url);
	});
	
	test('Must have a title equal to: "One Punch"', () => {
		const title = wrapper.find('p').text();
		expect(title).toBe(props.title)
	});

	test('The div element must has the css class: ".card"', () => {
		const div = wrapper.find('div');
		expect( div.exists('.card')).toBe(true);
	});
});
