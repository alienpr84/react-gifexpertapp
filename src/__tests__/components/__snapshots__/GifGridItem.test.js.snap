// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <GifGridItem /> Must show <GifGridItem /> component correctly 1`] = `
<div
  className="card"
>
  <img
    alt="One Punch"
    src="https://www.google.com"
  />
  <p>
    One Punch
  </p>
</div>
`;
